app/services/participant_outcomes/create.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module ParticipantOutcomes
+  class Create
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    STATES = %w[passed failed].freeze
+    COURSE_IDENTIFIERS = %w[npq-early-headship-coaching-offer npq-additional-support-offer].freeze
+    COMPLETION_DATE_FORMAT = /\d{4}-\d{2}-\d{2}/
+
+    attr_reader :created_outcome
+
+    attribute :lead_provider
+    attribute :participant
+    attribute :course_identifier
+    attribute :state
+    attribute :completion_date
+
+    validates :lead_provider, presence: true
+    validates :participant, presence: true
```

Comment:
When calling this service with `participant: nil` in the params I'm getting these error messages below. Think it should be another message maybe?

```
["Participant can't be blank",
 "The participant has not had a 'completed' declaration submitted for them. Therefore you cannot update their outcome."].
```

A bit different message when when course_identifier is nil as well:

```
service = ParticipantOutcomes::Create.new(
  lead_provider: Application.accepted.last.lead_provider,
  course_identifier: nil,
  participant: Application.accepted.last.user,
  completion_date: "2022-12-01",
  state: "passed",
)
service.valid?
service.errors.full_messages
```
```
["The participant has not had a 'completed' declaration submitted for them. Therefore you cannot update their outcome."]
```

Reply:
Ah, I didn't put a custom validation error in for the participant being blank as the service won't get called in that scenario from the controller. I'll add one in for completeness, same for `lead_provider`.

I couldn't replicate the issue with `course_identifier: nil`, I get the correct error (+ declarations error as the user has no declarations):

```
["Course identifier The entered '#/course_identifier' is not recognised for the given participant. Check details and try again.",
 "The participant has not had a 'completed' declaration submitted for them. Therefore you cannot update their outcome."]
```